### DISCUSSION TITLE
Set master switch in Virtual Chassis using API

### DISCUSSION THREAD

**Discussion by brianatlarge** [netbox-community/netbox#12744] [Help Wanted!] [answered]
I have a feeling I'm not doing something right. 

I have a virtual chassis with an ID of 103 and a name of "This-Vc".

I have a switch in the same site as the virtual chassis called "This-Sw".

I try to do a PATCH to /api/dcim/virtual-chassis/103 with this payload:

`{
    "master": {
        "name": "This-Sw"
    }
}`

I get a 200 response, but it still says "master" is set to null. What gives?

**Comment by jcollie** [comment]
Try setting it to just the device ID of the switch that you want to be master:

```
{"master": 245}
```

**Reply by brianatlarge** [reply]
ID of the device I want to be master is 820, but changing the payload to this exhibits the same behavior where nothing is changed.

`{
    "master": 820
}`

**Comment by brianatlarge** [comment, marked as answer]
Ok, I figured it out. I was missing the trailing / in my URL. The payload also needed to be:

```
{
    "master": {
        "name":"This.Sw"
    }
}
```
